modules/gem5/_posts/2000-03-01-assignment1.md

```diff
@@ -65,18 +61,11 @@ void multiply(double **A, double **B, double **C, int size)
 You can find the definitions for the workload objects in gem5 under `workloads/workloads.py`.
 In this assignment, we will only be using `MatMulWorkload`.
 In order to create an object of `MatMulWorkload` you just need to pass matrix size (an integer) `mat_size` to its constructor (`__init__`) function.
+
 **In your configuration choose an appropriate value for `mat_size`**. It should be large enough that it makes your workload interesting.
 Since changing `mat_size` will influence simulation time, as a guideline, choose a value that results in simulation times less than 10 minutes (hostSeconds < 600).
 We found that setting mat_size to 224 will result in a simulation time of around 5 minutes which is a reasonable compromise.
 
-**CAVEAT [PLEASE READ CAREFULLY]**: When using this workload with gem5, your simulation will output two sets of statistics in the same `stats.txt` file. Each set of statistics start with a line like below.
-
-```text
----------- Begin Simulation Statistics ----------
-```
-
-Please make sure to **ignore** the **second** set of generated statistics in your analysis.
```

Comment:
Can you explain why this is dropped? Did we fix the output somehow?

Reply:
I dropped the m5 instructions from the source code. I feel the students should not be worried about ROI in this assignment and the stats collected for this program should be for the entire matrix multiplication workload.

Reply:
Hmm... but then they will see the initialization instructions as well. In fact, I would bet initialization is going to dominate. Why do you think this is better?

Reply:
In this assignment, I felt that the students should only consider the entire program. I agree that the initialization will dominate. We introduce ROI in assignment 2 and I omitted it to maintain the flow.

Reply:
Ok, that sounds good to me.

Reply:
I thought about it for some time. I am examining the extent to which the initialization part dominates. I’m currently testing this and will revert soon.

Reply:
Okay, there are 15% -- 30% instructions in the initialization phase. We suggested using 224x224 matrices, where 14.4% of the instructions are in the init phase. I think we can go ahead with the current setup and introduce ROI in the next assignment.

Reply:
Sounds great. Thanks for looking into it.